### PR TITLE
[department-of-veterans-affairs/va-virtual-agent#873] Raises error if…

### PR DESCRIPTION
**Pull request by ksk385**
Raises error if LH call fails

Coauthored-by: swapna.krishnankutty@va.gov

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- If we get back non 200 code HTTP responses from LightHouse API, we want to log this error in Staging and also raise a Sentry alert

## Related issue(s)
[Issue Link](https://github.com/department-of-veterans-affairs/va-virtual-agent/issues/873)

## Testing done

- We tried to view the logs and only see that there were no Appeals. We need to be able to troubleshoot why the LH Sandbox call is not returning Appeals data in vets-api Staging

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
Only Virtual Agent (Chatbot)

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/controllers/v0/virtual_agent/virtual_agent_appeal_controller.rb [hunk 1]
```diff
@@ -37,6 +37,10 @@ def get_appeal_from_lighthouse(ssan, user_name)
         http.use_ssl = true
 
         response = http.request(req)
+        if response.code != '200'
+          Rails.logger.error "Lighthouse API returned #{response.code}: #{response.body}"
+          raise "#{response.code}: #{response.body}"
```
Comment (jeremy6d) [inline app/controllers/v0/virtual_agent/virtual_agent_appeal_controller.rb:42]: if the requirements are to raise this error only in staging, why wouldn't we simply add a `if Rails.env.staging?` on the end of the line?

**Review by jeremy6d — CHANGES_REQUESTED** [review]
I think we need to be explicit about when we want to raise errors vs. not raise errors. Raising an error unintentionally is a big risk in my estimation. Here's what I would suggest: 

1. being clear about the intention as close to the error being raised as possible
2. writing a test that exercises raising this error when `Rails.env.staging?` is stubbed true vs false

File: app/controllers/v0/virtual_agent/virtual_agent_appeal_controller.rb (hunk 1 above)
Comment (swapna-tw) [reply app/controllers/v0/virtual_agent/virtual_agent_appeal_controller.rb:42]: The highlighted part of code in VirtualAgentAppealController class, gets executed only when the function get_appeal_from_lighthouse is called and this function is called in line# 12 only if the environment is 'staging' based on line# 9 (as that is the only environment where Lighthouse's sandbox env is called to fetch appeals data for test users). 
In all other environments line# 16 and 17 are used, which do not use Rails.logger.error.

Comment (jeremy6d) [reply app/controllers/v0/virtual_agent/virtual_agent_appeal_controller.rb:42]: I understand your point that the code path prevents this exception from being raised for any reason other than the env being staging. 

In that case, I think the method `#get_appeal_from_lighthouse` should be renamed so it's explicit that it's a method for staging env only.

Comment (jeremy6d) [reply app/controllers/v0/virtual_agent/virtual_agent_appeal_controller.rb:42]: I also think we need to introduce at least a unit test for this new functionality. A unit test shows the next person who reads your code what you intended. I think my experience is emblematic of the confusion that is possible.

**Review by jeremy6d — CHANGES_REQUESTED** [review]
Sorry but I think these are important points that require changes. I'm happy to jump on a call if you think I'm not understanding things adequately.

**Review by jeremy6d — APPROVED** [review]
Great work, thanks for being so agreeable about my comments.

**Pull request merged**